File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/mapping/NotificationMapper.java
package com.github.arhor.simple.expense.tracker.service.mapping;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import com.github.arhor.simple.expense.tracker.data.model.Notification;
import com.github.arhor.simple.expense.tracker.model.NotificationDTO;

@Mapper(config = SharedMappingConfig.class)
public interface NotificationMapper {

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "timestamp", ignore = true)
    Notification mapDtoToEntity(NotificationDTO dto, Long targetUserId, Long sourceUserId);

    NotificationDTO mapProjectionToDto(Notification.CompactProjection projection);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/mapping/NotificationMapper.kt
package com.github.arhor.simple.expense.tracker.service.mapping

import com.github.arhor.simple.expense.tracker.data.model.Notification
import com.github.arhor.simple.expense.tracker.data.model.projection.CompactNotificationProjection
import com.github.arhor.simple.expense.tracker.model.NotificationDTO
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper(config = SharedMappingConfig::class)
interface NotificationMapper {

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "timestamp", ignore = true)
    fun mapDtoToEntity(dto: NotificationDTO, targetUserId: Long, sourceUserId: Long): Notification

    fun mapProjectionToDto(projection: CompactNotificationProjection): NotificationDTO
}
